cmd/codegen/util/go.go

```diff
@@ -168,3 +171,41 @@ func getModGraph(dir string) (string, error) {
 
 	return out, nil
 }
+
+func IsolatedGoPath() (string, error) {
+	configDir, err := ConfigDir()
+	if err != nil {
+		return "", errors.Wrapf(err, "getting JX_HOME")
+	}
+	path := filepath.Join(configDir, "codegen", "go")
+	err = os.MkdirAll(path, DefaultWritePermissions)
+	if err != nil {
+		return "", errors.Wrapf(err, "making %s", path)
+	}
+	return path, nil
+}
+
+func HomeDir() string {
+	if h := os.Getenv("HOME"); h != "" {
+		return h
+	}
+	h := os.Getenv("USERPROFILE") // windows
+	if h == "" {
+		h = "."
+	}
+	return h
+}
+
+func ConfigDir() (string, error) {
+	path := os.Getenv("JX_HOME")
```

Comment:
So per default, you want to install this into JX_HOME? Personally, I am no fan of that. It is not what I would expect. What's the driving reason for this change?

There is btw a clean target in the codegen Makefile which deletes the binaries. That probably would need to be looked at as well.

Reply:
FYI, my plan was to:

* make codegen a proper submodule with its own go.mod
* in the main build install and use a pinned version of codegen via `go get`
* adjust the make rules in the codgen Makefile to allow for development and use of the latest version

Reply:
> What's the driving reason for this change?

It doesn't work doing it in the gopath as you already have stuff cloned there onto branches - no way we should be reusing that.

Reply:
> There is btw a clean target in the codegen Makefile which deletes the binaries. That probably would need to be looked at as well.

Will do.

Reply:
> FYI, my plan was to:

> make codegen a proper submodule with its own go.mod
> in the main build install and use a pinned version of codegen via go get
> adjust the make rules in the codgen Makefile to allow for development and use of the latest version

That's orthogonal to this.

Reply:
I think with your "expect" thing, you are thinking as the developer of the feature, not the end user. As the end user I would expect these tools to work in isolation, not by fiddling with my development environment.

Reply:
> It doesn't work doing it in the gopath as you already have stuff cloned there onto branches - no way we should be reusing that.

Maybe. But then I'd choose a tmp directory which gets deleted afterwards and not ~/.jx

Reply:
> That's orthogonal to this.

trueish